### ISSUE TITLE
redis entries are not deleted

### ISSUE THREAD

**Issue by aks-zero** [davehorton/drachtio-siprec-recording-server#36]
We use this application with Freeswitch. Entries created in the redis server are not deleted and stay forever even after the SIPREC call is terminated with BYE. It seems it's happening all the time. Will appreciate any help for this.
Thanks
AKS

**Comment by davehorton** [comment]
The redis keys are inserted with an expiry of 10 seconds, [here](https://github.com/davehorton/drachtio-siprec-recording-server/blob/c050b20dfa8f6ec65ad8f71d9f1c3a53f62f0f07/lib/freeswitch-call-handler.js#L111)

Therefore, they should disappear after 10 seconds.  Perhaps the latter set request is somehow eliminating this expires value.  Not sure, I will have to look into it

**Comment by davehorton** [comment]
could you do a test for me?

Try changing [this line](https://github.com/davehorton/drachtio-siprec-recording-server/blob/c050b20dfa8f6ec65ad8f71d9f1c3a53f62f0f07/lib/freeswitch-call-handler.js#L167)

from 
```
.set(sessionId, sdp)
```
to 
```
.set(sessionId, sdp, 'KEEPTTL')
```
and see if that fixes it?

**Comment by aks-zero** [comment]
I guess that's the issue. I never see any TTL on the key as if it wasn't never set (it always returns -1). And that could be because the TTL is being discarded when the value for the key is SET second time.

Since the KEEPTTL is supported only in Redis >= 6.0, I couldn't test. Will try after upgrading Redis.

Alternatively, may be we can set the the expiry again during the second call using 'SET EX' option or 'SETEX' command.

**Comment by davehorton** [comment]
yes, you could just replicate the set ex command as used earlier.  Try that -- if it works let me know.  Feel free to make a PR

**Comment by aks-zero** [comment]
Yes, that worked. Thanks a lot.
`127.0.0.1:6379> KEYS *
1) "2fb147ed-549f-4ecf-aae0-0f9444cf9bb0"
127.0.0.1:6379> 
127.0.0.1:6379> 
127.0.0.1:6379> ttl 2fb147ed-549f-4ecf-aae0-0f9444cf9bb0
(integer) 3
127.0.0.1:6379> 
127.0.0.1:6379> ttl 2fb147ed-549f-4ecf-aae0-0f9444cf9bb0
(integer) 0
127.0.0.1:6379> 
127.0.0.1:6379> ttl 2fb147ed-549f-4ecf-aae0-0f9444cf9bb0
(integer) -2
127.0.0.1:6379> 
127.0.0.1:6379> KEYS *
(empty list or set)
127.0.0.1:6379> 
`
